internal/handlers/sqlite/msg_datasize.go

```diff
@@ -16,13 +16,96 @@ package sqlite
 
 import (
 	"context"
+	"fmt"
+	"time"
 
+	"github.com/FerretDB/FerretDB/internal/backends"
+	"github.com/FerretDB/FerretDB/internal/handlers/common"
+	"github.com/FerretDB/FerretDB/internal/handlers/commonerrors"
+	"github.com/FerretDB/FerretDB/internal/handlers/commonparams"
+	"github.com/FerretDB/FerretDB/internal/types"
+	"github.com/FerretDB/FerretDB/internal/util/lazyerrors"
 	"github.com/FerretDB/FerretDB/internal/util/must"
 	"github.com/FerretDB/FerretDB/internal/wire"
 )
 
 // MsgDataSize implements HandlerInterface.
 func (h *Handler) MsgDataSize(ctx context.Context, msg *wire.OpMsg) (*wire.OpMsg, error) {
-	// TODO https://github.com/FerretDB/FerretDB/issues/2775
-	return nil, notImplemented(must.NotFail(msg.Document()).Command())
+	document, err := msg.Document()
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	if err = common.Unimplemented(document, "keyPattern", "min", "max"); err != nil {
+		return nil, err
+	}
+
+	common.Ignored(document, h.L, "estimate")
+
+	var namespaceParam any
+
+	if namespaceParam, err = document.Get(document.Command()); err != nil {
+		return nil, err
+	}
+
+	namespace, ok := namespaceParam.(string)
+	if !ok {
+		return nil, commonerrors.NewCommandErrorMsgWithArgument(
+			commonerrors.ErrBadValue,
+			fmt.Sprintf("collection name has invalid type %s", commonparams.AliasFromType(namespaceParam)),
+			document.Command(),
+		)
+	}
+
+	dbName, cName, err := commonparams.SplitNamespace(namespace, document.Command())
+	if err != nil {
+		return nil, err
+	}
+
+	db, err := h.b.Database(dbName)
+	if err != nil {
+		if backends.ErrorCodeIs(err, backends.ErrorCodeDatabaseNameIsInvalid) {
+			msg := fmt.Sprintf("Invalid database specified '%s'", dbName)
+			return nil, commonerrors.NewCommandErrorMsgWithArgument(commonerrors.ErrInvalidNamespace, msg, document.Command())
+		}
+
+		return nil, lazyerrors.Error(err)
+	}
+	defer db.Close()
+
+	c, err := db.Collection(cName)
+	if err != nil {
+		if backends.ErrorCodeIs(err, backends.ErrorCodeCollectionNameIsInvalid) {
+			msg := fmt.Sprintf("Invalid collection name: %s", cName)
+			return nil, commonerrors.NewCommandErrorMsgWithArgument(commonerrors.ErrInvalidNamespace, msg, document.Command())
+		}
+
+		return nil, lazyerrors.Error(err)
+	}
+
+	started := time.Now()
+
+	stats, err := c.Stats(ctx, new(backends.CollectionStatsParams))
+	if backends.ErrorCodeIs(err, backends.ErrorCodeDatabaseDoesNotExist) ||
```

Comment:
Is it a possible case? I don't see `Stats` can return `ErrorCodeDatabaseDoesNotExist` or `ErrorCodeCollectionDoesNotExist`.

Reply:
or is it a different stats?.. I think I'm lost :)

Reply:
It's a `Collection.Stats` which returns error. The newly added interface is for `Backend.Stats`.

This makes me think, perhaps we can rename them for clarity like below?
`Collection.Stats` -> `Collection.CollectionStats`
`Database.Stats` -> `Database.DatabaseStats` 
`Backend.Stats` -> `Backend.ServerStats`

Reply:
`Backend.ServerStats` is in fact `Backend.ServerStatus`, right? As it doesn't have the same fields as `CollectionStats` and `DatabaseStats`?

Reply:
Yes sure `Backend.ServerStatus` is more accurate :pray:

Reply:
And I think for collections and databases I like the current naming - `Collection.Stats` and `Database.Stats` as we don't duplicate things in naming :) It's a bit confusing in the beginning, but I think it's fine.

Reply:
Yes true, lets leave those naming as it was :+1: